### PR TITLE
Enable building the JIT on OSX

### PR DESCRIPTION
**Pull request by nbhuiyan**
Signed-of-by: Nazim Uddin Bhuiyan <nazimudd@ualberta.ca>

### CODE REVIEW

**Comment by nbhuiyan** [comment]
@DanHeidinga @0xdaryl requesting review for this changeset.

While this is not going to successfully build all of the JIT on OSX (due to issues such as assembly files and [BlockContainerAllocator](https://github.com/eclipse/openj9/blob/5e482173e62fc430ddcfadcbe57c0b6df04e8202/runtime/compiler/optimizer/JProfilingBlock.hpp#L40) not being const-correct), I would like these changes to go through.

**Comment by DanHeidinga** [comment]
Can you split the j9sock.c and zipsup.c changes out of this PR?  They're both include in #2261 which is being reviewed separately.

File: runtime/compiler/runtime/J9ValueProfiler.hpp [hunk 1]
```diff
@@ -68,6 +68,14 @@ typedef uint32_t ProfileAddressType;
 typedef uintptr_t ProfileAddressType;
 #endif
 
+/** 
+ * Clang does not consider uintptr_t to be equivalent to uint64_t
+ */
+
+#if defined(OSX)
```
Comment (DanHeidinga) [inline runtime/compiler/runtime/J9ValueProfiler.hpp]: Can this be merged into the if / else logic above?  This will typedef ProfileAddressType twice.

**Comment by nbhuiyan** [comment]
j9sock.c and zipsup.c have been removed from this PR. Requesting another review.

**Comment by 0xdaryl** [comment]
CI failures are not related to this PR (.gitattributes issue).  Merging.

**Pull request merged**